### PR TITLE
Strengthen instruction operand types

### PR DESCRIPTION
**Pull request by Kha**
By the power of static typing!

Fixes using stack parameters for indexing, comparisons, and additions

### CODE REVIEW

**Comment by Kha** [comment]
PhiGenerator left out for now

File: src/main/scala/mjis/RegisterAllocator.scala [hunk 1]
```diff
@@ -440,8 +440,8 @@ class FunctionRegisterAllocator(function: AsmFunction,
         instr.operands(idx) = map(r)
       case (a: AddressOperand, idx) =>
         instr.operands(idx) = a.copy(
-          base = a.base.flatMap(r => Some(map(r.asInstanceOf[RegisterOperand]))),
-          offset = a.offset.flatMap(r => Some(map(r.asInstanceOf[RegisterOperand])))
+          base = a.base.map(map(_).asInstanceOf[RegisterOperand]),
```
Comment (Kha) [inline src/main/scala/mjis/RegisterAllocator.scala]: @jspam Does the allocator actually guarantee this is not a spilled AddressOperand?

Comment (jspam) [inline src/main/scala/mjis/RegisterAllocator.scala]: Yes, it currently reloads all operands into a register before their use. (One planned optimization is to use the `OperandSpec`s to allow some spilled operands to be loaded directly from memory.)

**Comment by Kha** [comment]
Superseded by #302 (who needs static typing, anyways)

**Pull request closed without merging**